### ISSUE TITLE
Launcher default apps intent opens wrong screen on HUAWEI Mate 10 Lite

### ISSUE THREAD

**Issue by d4rken** [HIK-apps/android-dev-launcher-changelog#3]
Android 8

This is the screen we get to:
[Screenshot_20230821_191514](https://github.com/HIK-apps/android-dev-launcher-releases/assets/1439229/ad434982-8beb-40ca-a764-3dff87a12bba)

and this is the one where it can be changed:
https://www.youtube.com/watch?v=4dIsi0q2Nes

The screen we need to get to is:
```
mResumedActivity: ActivityRecord{78697f6 u0 com.android.settings/.Settings$AppAndNotificationDashboardActivity t70}
```

Can debug this further if you give me source-code access. Really love your apps, would love to tweak little things for my workflow :beers:

**Comment by blipinsk** [comment]
> Can debug this further if you give me source-code access. Really love your apps, would love to tweak little things for my workflow 🍻

I love your enthusiasm, but unfortunately, I'm not planning to open-source it (or introduce new developers to the project, even ones as awesome as you 😢).

I would love to tweak those things for you though! Those could be beneficial for the entire Android Dev Launcher community.

--

So you're saying that you're seeing the incorrect screen after tapping `Launcher` button?
Can you help me check if there's a `Settings.ACTION_<XYZ>` value which represents the screen you're referring to?

**Comment by d4rken** [comment]
> Can you help me check if there's a Settings.ACTION_<XYZ> value which represents the screen you're referring to?

Which settings value are you currently using?

**Comment by blipinsk** [comment]
If it's Android 8, it should be `Settings.ACTION_MANAGE_DEFAULT_APPS_SETTINGS`.
We might be getting into some Huawei-specific behavior. I have tested this on multiple systems (I believe Android 8 as well) and it was working correctly.

**Comment by d4rken** [comment]
The intent suggested here works for me:

https://stackoverflow.com/a/54191831/1251958

Directly opens the default apps screen:
[Screenshot_20230821_202822](https://github.com/HIK-apps/android-dev-launcher-releases/assets/1439229/8875fef8-b1fb-4c85-b0fd-d00d43b60d1b)

What about:

* Check whether it's a Huawei device
* If so check if any apps can handle `com.android.settings.PREFERRED_SETTINGS`
* If yes send it (but catch Exceptions, there might be `ActivityNotFound` or `SecurityException`)
* If there is no such activity or an error happens, then fall back to the current intent.

**Comment by blipinsk** [comment]
Yes, I can totally do that. It actually fits with all my "fallback"-mechanisms quite well, I just need to add an `isHuawei` check.

Random question, is this your only Huawei device? Maybe you've got something with another OS version as well? I'd be mostly interested in:
* 7 or 9, 
* 10+ (which is much more complex than just sending the `ACTION_MANAGE_DEFAULT_APPS_SETTINGS`)

**Comment by d4rken** [comment]
> Random question, is this your only Huawei device? Maybe you've got something with another OS version as well? I'd be mostly interested in:

Unfortunately not, sorry. The only other Huawei device I currently have here runs Android 2.3 :grin:.

I've set your launcher already up on a few different ROM types (MIUI, OxygenOS etc.). So far this one was the only one that was problematic. If I find other I'll probably raise another issue ticket :beers:.

Offtopic too, FYI: There was one device, either a OnePlus Nord2 5G or Redmi A2, where ADB didn't have enough permissions to grant the WRITE_SECURE_SETTINGS. The device was rooted though, and that worked then. Not much you can do about that though :shrug:
